lua/telescope/previewers.lua

```diff
@@ -21,7 +21,24 @@ Previewer.__index = Previewer
 local bat_options = {"--style=plain", "--color=always", "--paging=always"}
 local has_less = (vim.fn.executable('less') == 1) and config.values.use_less
 
+local function_preview_director = function(filename)
+  if 1 == vim.fn.executable('file') then
+    local handle = io.popen('file --mime-type -b ' .. filename)
+    local mime_type = vim.fn.split(handle:read('*a'), '\n')[1]
+    handle:close()
```

Comment:
Let's make a util function for executing command and not having to remember to close handles.

Reply:
Done. Also used this new function for a `popen` call in `builtin.man_pages`